signer/src/block_observer.rs

```diff
@@ -149,7 +150,9 @@ where
         known_deposit_requests: &DepositRequestMap,
         block: bitcoin::Block,
     ) -> Result<(), error::Error> {
-        let stacks_blocks = self.stacks_client.get_last_tenure_blocks().await?;
+        let info = self.stacks_client.get_tenure_info().await?;
+        let stacks_blocks =
+            fetch_unknown_ancestors(&self.stacks_client, &self.storage, info.tip_block_id).await?;
```

Comment:
...and add the module namespace to this call:

```suggestion
            stacks_api::fetch_unknown_ancestors(&self.stacks_client, &self.storage, info.tip_block_id).await?;
```